app/src/main/java/com/islamversity/reyan/service/MyFirebaseMessagingService.kt

```diff
@@ -0,0 +1,101 @@
+package com.islamversity.reyan.service
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.media.RingtoneManager
+import android.os.Build
+import android.os.Bundle
+import androidx.core.app.NotificationCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.islamversity.reyan.MainActivity
+import com.islamversity.reyan.R
+import com.islamversity.reyan.service.NotificationDataType.Companion.NOTIFICATION_DATA_KEY
+import timber.log.Timber
+
+
+class MyFirebaseMessagingService: FirebaseMessagingService() {
+
+
+    override fun onNewToken(token: String) {
+        Timber.d("Notification : Refreshed token: $token")
+        sendFCMTokenToServer(token)
+    }
+
+    private fun sendFCMTokenToServer(token: String) {
+        // Cache.set(CacheKeys.Firebase.FCM_TOKEN_FIREBASE, token)
+        Timber.d("Notification : sendRegistrationTokenToServer($token)")
+    }
+
+    override fun onMessageReceived(remoteMessage: RemoteMessage) {
+        Timber.d("Notification : From: ${remoteMessage.from}")
+
+        val data  = remoteMessage.data
+
+        // Check if message contains a data payload.
+        if (data.isNotEmpty()) {
+            Timber.d("Notification : Message data payload: $data")
+
+            val bundle = Bundle()
```

Comment:
I would suggest let's create a parcelable `Model`s to pass each notification data?

Reply:
why ? what is it usefull for ?

Reply:
passing a bundle is not really a safe way of communication
testing wise it's much harder
also Models provide a concrete communication model.
please check our `LocalModels` which can be done using a bundle but the models make it easier.

maintenance is also much easier using models

Reply:
we do  not know what comes. how to use  a model for different  notifications ?

Reply:
we should enforce that